### PR TITLE
support push down projection

### PR DESCRIPTION
**Pull request by fzhedu**
support pushing down projection.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+.idea
```
Comment (hanfei1991) [inline .gitignore:1]: remove this?

Comment (fzhedu) [reply .gitignore:1]: it is added by defalut when using CLion, so add this is better.

**Comment by windtalker** [comment]
LGTM

**Review by hanfei1991 — APPROVED** [review]
LGTM

**Pull request merged**